### ISSUE TITLE
Follow-up plan regarding changes to resolvers

### ISSUE THREAD

**Issue by odelalleau** [omry/omegaconf#535]
#445 introduces several changes to resolvers that triggered multiple discussions on related follow-up changes. Here is an overview of what is planned once #445 is merged:

- [x] Address #488 by validating & converting the result of all interpolations (both node interpolations and resolver outputs) when the target node is typed
- [x] Do a pass on the notebook to make sure it is in synch with the doc, and works with Binder (#544)
- [x] Change the `env` resolver so that it does not parse environment variables anymore, instead keeping them as strings (that may be cast into a different type thanks to the previous bullet point) -- will also resolve #383
- [x] Add support for resolver namespaces (ex: `oc.env`) and deprecate the plain `env` resolver (#573)
- [x] Add new resolver `oc.decode` to parse strings (ex: those output by the `oc.env` resolver) (#574)
- [x] Improve API of `register_new_resolver()` according to #370
- [x] Deprecate `get_resolver()` (#608)
- [X] Forbid whitespace characters around dots in node interpolations and resolver names (#594)
- [x] Deprecate `OmegaConf.is_none()` (#547)
- [x] Implement new resolver cache behavior as discussed in #526 
- [x] Add support for $ in key names (see https://github.com/facebookresearch/hydra/issues/1437#issuecomment-789977874) => #600 (OmegaConf) / https://github.com/facebookresearch/hydra/pull/1605 (Hydra)
- [x] Fix issue with handling of quoted strings (#617)
- [x] Add resolvers `oc.dict.values` and `oc.dict.keys` (#643)
- [x] Add resolver `oc.create` (#645)
- [x] Use getitem syntax as per #179
- [X] SImplify the escaping of quotes (#695)
- [x] Simplify escaping further (#708)
- [x] Fix crash when escaped interpolations do not follow the expected interpolation syntax (#666)
- [x] Port corresponding changes to Hydra (https://github.com/facebookresearch/hydra/pull/1598 / https://github.com/facebookresearch/hydra/pull/1599 / https://github.com/facebookresearch/hydra/pull/1601 / https://github.com/facebookresearch/hydra/pull/1605)
- [x] Address thread-safety concern related to parsing of interpolations (#715)

Documentation (to be done in Q3'21):
- [x] Document OmegaConf's grammar (when doing so, also take care of https://github.com/omry/omegaconf/pull/677#discussion_r614452526)
- ~[ ] Update OmegaConf's slide deck~ (will transfer task to someone else)
- ~[ ] Create a dedicated doc page for interpolations (https://github.com/omry/omegaconf/pull/578#discussion_r590793981)~ (will transfer task to someone else)
- [X] Create a dedicated doc page for built-in resolvers (https://github.com/omry/omegaconf/pull/644#issuecomment-806189169) (done as part of #679)

**Comment by odelalleau** [comment]
Closing as everything is either done or will be transferred to someone else.
